src/pages/BookDetail/BookDetail.tsx

```diff
@@ -0,0 +1,131 @@
+import { useEffect, useState } from 'react';
+import { useParams, useNavigate } from 'react-router-dom';
+import ArrowBackIosRoundedIcon from '@mui/icons-material/ArrowBackIosRounded';
+
+import { Layout } from '../../components/Layout/Layout';
+import { BookCard } from '../../components/BookCard/BookCard';
+import mockbooks from '../../data/mockbooks.json';
+import reviews from '../../data/reviews.json';
+import { Book, Review } from '../../../types/types';
+import style from './BookDetail.module.scss';
+
+export const BookDetail: React.FC = () => {
+  const { id } = useParams();
+  const navigate = useNavigate();
+  const [book, setBook] = useState<Book | null>(null);
+  const [bookReviews, setBookReviews] = useState<Review[]>([]);
+  const books: Book[] = mockbooks;
+
+  useEffect(() => {
+    const foundBook = books.find((b) => b.id === Number(id));
+    if (foundBook) {
+      setBook(foundBook);
+      const bookSpecificReviews = reviews.filter(
+        (review) => review.bookId === Number(id),
+      );
+      setBookReviews(bookSpecificReviews);
+    }
+  }, [id]);
```

Comment:
```suggestion
  useEffect(() => {
    const foundBook = books.find((b) => b.id === Number(id));
    
    if (foundBook) {
      setBook(foundBook);
      
      const bookSpecificReviews = reviews.filter(
        (review) => review.bookId === Number(id),
      );
      setBookReviews(bookSpecificReviews);
    }
  }, [id]);
```

separate logical/instruction groups by new line